benchmark/benchmark.R

```diff
@@ -0,0 +1,179 @@
+  
+  all_proteins <- c(union(unique_ecoli_proteins, unique_yeast_proteins))  # find out the significant proteins in FragData
+  
+  extracted_proteins <- sapply(all_proteins, function(x) {
+    split_string <- strsplit(x, "\\|")[[1]]  # Split the string by '|'
+    if (length(split_string) >= 2) {
+      return(split_string[2])  # Return the second element
+    } else {
+      return(NA)  # Return NA if there's no second element
+    }
+  })
+  
+  extracted_proteins <- unname(unlist(extracted_proteins))
+  
+  proteins <- c(extracted_proteins)
+  
+  
+  TP <- comparisonResult %>% filter(grepl(paste(proteins, collapse = "|"), Protein) & adj.pvalue < 0.05) %>% nrow()
+  
+  
+  FP <- comparisonResult %>% filter(!grepl(paste(proteins, collapse = "|"), Protein) & adj.pvalue < 0.05) %>% nrow()
+  
+  
+  TN <- comparisonResult %>% filter(!grepl(paste(proteins, collapse = "|"), Protein) & adj.pvalue >= 0.05) %>% nrow()
+  
+  
+  FN <- comparisonResult %>% filter(grepl(paste(proteins, collapse = "|"), Protein) & adj.pvalue >= 0.05) %>% nrow()
+  
+  cat("True Positives (Yeast and EColi): ", TP, "\n")
+  cat("False Positives (Human Samples)", FP, "\n")
+  cat("True Negatives", TN, "\n")
+  cat("False Negatives", FN, "\n")
+  
+  FPR <- FP / (FP + TN)
+  
+  # Accuracy
+  accuracy <- (TP + TN) / (TP + TN + FP + FN)
+  
+  # Recall
+  recall <- TP / (TP + FN)
+  
+  results <- data.frame(
+    Label = label,
+    TP = TP,
+    FP = FP,
+    TN = TN,
+    FN = FN,
+    FPR = FPR,
+    Accuracy = accuracy,
+    Recall = recall
+  )
+  
+  return(results)
+  
+}
+
+start_time <- Sys.time()
+
+# Use fread directly to read the CSV
+fragpipe_raw = data.table::fread("..//data//FragPipeMsStatsBenchmarking.csv")
+
+head(fragpipe_raw)
+
+fragpipe_raw$Condition = unlist(lapply(fragpipe_raw$Run, function(x){
+  paste(str_split(x, "\\_")[[1]][4:5], collapse="_")
+}))
+
+fragpipe_raw$BioReplicate = unlist(lapply(fragpipe_raw$Run, function(x){
+  paste(str_split(x, "\\_")[[1]][4:7], collapse="_")
+}))
+
+# Convert to MSstats format
+msstats_format = MSstatsConvert::FragPipetoMSstatsFormat(fragpipe_raw, use_log_file = FALSE)
+
+
+# Define the tasks with descriptive labels
+data_process_tasks <- list(
+  list(
+    label = "Data process with Normalized Data",
+    result = function() dataProcess(msstats_format, featureSubset = "topN", n_top_feature = 20)
+  ),
+  list(
+    label = "Data process with Normalization and MBImpute False",
+    result = function() dataProcess(msstats_format, featureSubset = "topN", n_top_feature = 20, MBimpute = FALSE)
+  ),
+  list(
+    label = "Data process without Normalization",
+    result = function() dataProcess(msstats_format, normalization = "FALSE", n_top_feature = 20)
+  ),
+  list(
+    label = "Data process without Normalization with MBImpute False",
+    result = function() dataProcess(msstats_format, normalization = "FALSE", n_top_feature = 20, MBimpute = FALSE)
+  )
```

Comment:
Does `n_top_feature` parameter need to be initialized to anything here?  i thought it's only needed if featureSubset = "topN"